### PR TITLE
nrf_security: openthread config for mbedtls debug

### PR DESCRIPTION
**Pull request by pjasinski990**
Adds config that makes it possible to enable mbedtls debug configs with openthread. Currently it's impossible to select MBEDTLS_DEBUG if not using CUSTOM_OPENTHREAD_SECURITY (as it depends on MBEDTLS, which is promptless).

### CODE REVIEW

File: samples/openthread/cli/overlay-logging.conf [hunk 1]
```diff
@@ -37,3 +37,7 @@ CONFIG_LOG_BACKEND_RTT=y
 # Disable UART logging backend
 # CONFIG_LOG_BACKEND_UART=n
 CONFIG_SHELL_LOG_BACKEND=n
+
+# mbedTLS logs
```
Comment (maciejbaczmanski) [inline samples/openthread/cli/overlay-logging.conf:41]:
```suggestion
# Enable Mbed TLS logging
```
Also shouldn't we move it a bit higher in the file? maybe before `# Enable OpenThread logging` ?

**Review by maciejbaczmanski — COMMENTED** [review]
Thanks!
One comment, but it applies for all `.conf` files

**Review by maciejbaczmanski — APPROVED** [review]

**Review by edmont — APPROVED** [review]

**Review by canisLupus1313 — APPROVED** [review]

File: subsys/nrf_security/Kconfig.tls [hunk 2]
```diff
@@ -439,3 +439,9 @@ endmenu
 
 
 endif # MBEDTLS_TLS_LIBRARY
+
+config OPENTHREAD_MBEDTLS_DEBUG
```
Comment (joerchan) [inline subsys/nrf_security/Kconfig.tls:443]: Move this to "subsys/net/openthread" please.

**Review by joerchan — CHANGES_REQUESTED** [review]

File: subsys/net/openthread/Kconfig [hunk 3]
```diff
@@ -52,6 +52,12 @@ config OPENTHREAD_NRF_SECURITY
 config OPENTHREAD_MBEDTLS_LIB_NAME
 	default "mbedtls_external" if OPENTHREAD_NRF_SECURITY
 
+config OPENTHREAD_MBEDTLS_DEBUG
+	bool "MbedTLS logs for OpenThread"
+	select MBEDTLS
```
Comment (joerchan) [inline subsys/net/openthread/Kconfig:57]: Nit: Maybe a bit much for MBEDTLS_DEBUG to enable the MBEDTLS feature itself.

This is already done by OPENTHREAD_MBEDTLS, I'm thinking that should be enough.

**Review by joerchan — APPROVED** [review]

File: subsys/net/openthread/Kconfig (hunk 3 above)
Comment (pjasinski990) [reply subsys/net/openthread/Kconfig:57]: The problem is we use OPENTHREAD_NRF_SECURITY (not OPENTHREAD_MBEDTLS). MBEDTLS is not selected in our default configuration.

**Pull request merged**